src/data_types/w3c/credential.rs

```diff
@@ -73,356 +55,120 @@ pub struct CredentialSubject {
-            type_: CredentialStatusType::AnonCredsCredentialStatusList2023,
-            id,
-        }
-    }
-}
-
-#[derive(Debug, Clone, Default, PartialEq, Eq, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct CredentialSchema {
-    #[serde(rename = "type")]
-    pub type_: CredentialSchemaType,
-    pub definition: CredentialDefinitionId,
-    pub schema: SchemaId,
-    #[serde(default)]
-    pub encoding: CredentialValuesEncoding,
-}
-
-impl CredentialSchema {
+impl W3CCredential {
     pub fn new(
-        schema: SchemaId,
-        definition: CredentialDefinitionId,
-        encoding: CredentialValuesEncoding,
-    ) -> CredentialSchema {
-        CredentialSchema {
-            type_: CredentialSchemaType::AnonCredsDefinition,
-            definition,
-            schema,
-            encoding,
+        issuer: IssuerId,
+        attributes: CredentialAttributes,
+        proof: DataIntegrityProof,
+        version: Option<&VerifiableCredentialSpecVersion>,
+    ) -> Self {
+        let time = Utc::now();
+        let version = version.cloned().unwrap_or_default();
+        let (issuance_date, valid_from) = match version {
+            VerifiableCredentialSpecVersion::V1_1 => (Some(time), None),
+            VerifiableCredentialSpecVersion::V2_0 => (None, Some(time)),
+        };
+        Self {
+            context: Contexts::get(&version),
+            type_: ANONCREDS_CREDENTIAL_TYPES.clone(),
+            issuance_date,
+            valid_from,
+            issuer,
+            credential_subject: CredentialSubject {
+                id: None,
+                attributes,
+            },
+            proof: OneOrMany::Many(vec![CredentialProof::DataIntegrityProof(proof)]),
+            id: None,
         }
     }
-}
-
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub enum CredentialSchemaType {
-    #[serde(rename = "AnonCredsDefinition")]
-    AnonCredsDefinition,
-}
-
-impl Default for CredentialSchemaType {
-    fn default() -> Self {
-        CredentialSchemaType::AnonCredsDefinition
-    }
-}
 
-impl W3CCredential {
-    pub fn new() -> W3CCredential {
+    pub(crate) fn derive(
+        attributes: CredentialAttributes,
+        proof: DataIntegrityProof,
+        credential: &W3CCredential,
+    ) -> W3CCredential {
         W3CCredential {
-            context: ANONCREDS_CONTEXTS.clone(),
-            type_: ANONCREDS_CREDENTIAL_TYPES.clone(),
-            issuance_date: Utc::now(),
-            proof: OneOrMany::Many(Vec::new()),
-            ..Default::default()
+            context: credential.context.clone(),
+            type_: credential.type_.clone(),
+            issuer: credential.issuer.clone(),
+            id: credential.id.clone(),
+            issuance_date: credential.issuance_date,
+            valid_from: credential.valid_from,
+            credential_subject: CredentialSubject {
+                id: credential.credential_subject.id.clone(),
```

Comment:
Is this method used to create a derived credential? I don't think we always want to include the `id` property in a derived credential? Or maybe I'm misunderstanding this code incorrectly.

Reply:
Correct.   This method is used to construct a credential which will be put into a presentation.

Reply:
Then we should remove the `id` property I think, otherwise it reveals a correlatable identifier. The `id` property should only be revealed when it's revealed in the crednetial.